### PR TITLE
Add RDMA verbs configuration

### PR DESCRIPTION
**Pull request by dariavel**
RDAM configuration is now possible with the following environment variables:  RDMA_DEVICE, RDMA_DEVICE_PORT, RDMA_GID_INDEX,  RDMA_PKEY, RDMA_QUEUE_DEPTH, RDMA_TIMEOUT, RDMA_RETRY_CNT, RDMA_SL, RDMA_MTU and RDMA_TRAFFIC_CLASS.

### CODE REVIEW

**Comment by dariavel** [comment]
I signed it!

Regards,
Daria Velikovsky
SW Engineer, SW
Mellanox Technologies
Office: +972-747238531, Mobile: +972-543271181
www.mellanox.com<http://www.mellanox.com/>
Follow us on Facebook<http://www.facebook.com/pages/Mellanox-Technologies/223164879116> and Twitter<http://twitter.com/mellanoxtech>



________________________________

**Comment by noaezra** [comment]
I signed it!

**Comment by junshi15** [comment]
It looks good to me.

**Comment by dariavel** [comment]
All the authors have signed the CLA.

**Comment by noaezra** [comment]
I'm ok okay with my commits being contributed to this project.

File: tensorflow/contrib/verbs/rdma.cc
Comment (byronyi) [inline tensorflow/contrib/verbs/rdma.cc]: This line is way too long. Please wrap it to 80 columns.

**Comment by shamoya** [comment]
Hi @poxvoculi @jhseu, 
Have you add the chance to take a look ?

File: tensorflow/contrib/verbs/rdma.cc
Comment (poxvoculi) [inline tensorflow/contrib/verbs/rdma.cc]: Google style prefers:
} else {
  // set default device

File: tensorflow/contrib/verbs/rdma.cc
Comment (poxvoculi) [inline tensorflow/contrib/verbs/rdma.cc]: This control flow is a little odd, with the return inside the for loop and below at line 215.  How about:
port_num = 0;  // line 181.
for (port_index = 1;...
   if (port_attr.state == IBV_PORT_ACTIVE) {
       CHECK_EQ(0, port_num) << "Multiple active ports";
        port_num = port_index;
    }
}
CHECK_GT(port_num, 0) << "No active ports";

**Review by poxvoculi — APPROVED** [review]

**Comment by shamoya** [comment]
Thanks @poxvoculi, can we merge this ?

**Comment by shamoya** [comment]
@poxvoculi, we want to rearrange the commits of the PR (all of the commits are duplicated, probably something with the merges got mixed up). We will fix it tomorrow plz do not merge if you see this.

**Review by drpngx — CHANGES_REQUESTED** [review]
Block until refactor, as per comment.

**Comment by shamoya** [comment]
@drpngx @poxvoculi @jhseu - We fixed the duplicated commits and now all looks good.
No change in code since @poxvoculi review.
Thanks @noaezra @dariavel for the great work.

**Comment by shamoya** [comment]
@drpngx ?

**Review by drpngx — APPROVED** [review]

**Comment by drpngx** [comment]
Sounds good, let's see if the test suite passes.

**Comment by drpngx** [comment]
Could you pull rebase and push again? The different commits appear to be foiling the CLA bot.

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by shamoya** [comment]
Thanks @gunan @drpngx 
I guess the cla/google is fooled by - commits were authored by someone other than the pull request submitter ? 
If so, rebasing on top master on this PR will help ? 
Should we post a new PR with the changes ?

**Comment by drpngx** [comment]
Yes rebase usually works

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by gunan** [comment]
Looks like all the authors did sign the CLA, but CLA bot is confused because of multiple commits from different authors. Once the tests pass, I can merge this PR.

**Comment by shamoya** [comment]
@gunan , looks like there problem with the CI, not related to this patch.
Can we re-run the jenkins ?

**Comment by gunan** [comment]
You are right. I think last night we ran out of daily quota on the backend, it should be fixed now.

**Pull request merged**
